### PR TITLE
[SYCL] Update documents with recent project changes

### PR DESCRIPTION
**Pull request by vladimirlaz**
- Add development process description to contributing guide
 - Add info about skipping LIT tests targetting OpenCL device
   if corresponding RT is missed.

Signed-off-by: Vladimir Lazarev <vladimir.lazarev@intel.com>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -64,3 +64,30 @@ Then you just add a line to every git commit message:
...
 
 If you set your `user.name` and `user.email` git configs, you can sign your commit automatically with `git commit -s`.
+
+## Contribution process
+
+### Development
+
+ - Create a personal fork of the project on GitHub.
+ - Use **sycl** branch as baseline for your changes.
+ - Implement/fix your feature (follow [LLVM coding standards](https://llvm.org/docs/CodingStandards.html)).
```
Comment (Fznamznon) [inline CONTRIBUTING.md]: Fix your feature seems strange for me :) It look like that we can obviously commit feature which is not working.
How about "Prepare your patch"?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -64,3 +64,30 @@ Then you just add a line to every git commit message:
...
+ - Approval is lost once commit is changed. New approve and checks rerun required.
+ - Once approve is received and all checks pass pull request is ready to merge.
+
+### Merge
+
+Merge of pull request is done only by project maintainers. There are three options:
+ - [Rebase and merge] Is preferable choice for PR containing single commit.
+ - [Squash and merge] Is used when there are multiple commits in PR. Sqwashing is
+   done to shottern history and make sure that project is buildable on any commit.
+ - [Create a merge commit] Is used for pull down PRs to avoid duplication of
```
[the remark this answers is not in the document]
Comment (vladimirlaz) [reply CONTRIBUTING.md]: Initial version looks better because button is object but not subject.

File: CONTRIBUTING.md (hunk 1 above)
Comment (vladimirlaz) [inline CONTRIBUTING.md]:
```suggestion
 - Prepare your patch (follow [LLVM coding standards](https://llvm.org/docs/CodingStandards.html)).
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -64,3 +64,30 @@ Then you just add a line to every git commit message:
...
+     - **sycl-ubu-x64-pr** - runs all LIT tests on machine with OpenCL runtimes for CPU and GPU.
+ - Approval is lost once the PR branch is updated. New approval and checks rerun are required.
+ - Once approval is received and all checks pass, the pull request is ready to be merged.
+
+### Merge
+
+Merge of pull request is done only by project maintainers. There are three options:
+ - [Rebase and merge] The preferable choice for PRs containing a single commit.
+ - [Squash and merge] Used when there are multiple commits in the PR. Squashing is
+   done to shorten history and make sure that the project is buildable on any commit.
```
Comment (Fznamznon) [inline CONTRIBUTING.md]: I think sentence with "Squashing is done.." will look better in separate sub-bullet.

Comment (vladimirlaz) [reply CONTRIBUTING.md]: the sentence is note for the bullet. so it is reasonable to have it under the same bullet.

Comment (Fznamznon) [reply CONTRIBUTING.md]: This bullet looks to long. I suggest adding sub-bullet like this:
* [Squash and merge] Used when there are multiple commits in the PR. 
   * Squashing is done done to shorten history and make sure that the project is buildable on any commit.

Why not?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -64,3 +64,30 @@ Then you just add a line to every git commit message:
...
+
+ - Create a personal fork of the project on GitHub.
+ - Use **sycl** branch as baseline for your changes.
+ - Prepare your patch (follow [LLVM coding standards](https://llvm.org/docs/CodingStandards.html)).
+ - Build the project and run all tests (see [GetStartedWithSYCLCompiler.md](https://github.com/intel/llvm/blob/sycl/sycl/doc/GetStartedWithSYCLCompiler.md))
+
+### Review and acceptance testing
+
+ - Create a pull request for your changes following [Creating a pull request instructions](https://help.github.com/articles/creating-a-pull-request/).
+ - Once the pull request is approved by an Intel representative, testing is started.
```
Comment (bader) [inline CONTRIBUTING.md]: Please, make it clear that **check_pr** is started w/o "Intel representative" approval.

**Pull request merged**